sklearn/naive_bayes.py

```diff
@@ -39,6 +39,83 @@
 class _BaseNB(ClassifierMixin, BaseEstimator, metaclass=ABCMeta):
     """Abstract base class for naive Bayes estimators"""
 
+    _parameter_constraints: dict = {
+        "priors": ["array-like", StrOptions({"uniform", "empirical"}), None],
+    }
+
+    def __init__(self, *, priors=None):
+        self.priors = priors
+
+    def _validate_priors(self):
+        """Helper function to deprecate `class_prior` and `fit_prior` in favor
+        of `priors` for a consistent API.
+        """
+        if not (hasattr(self, "class_prior") or hasattr(self, "fit_prior")):
+            # consistent API with early return
+            self._priors = "empirical" if self.priors is None else self.priors
+            return
+
+        deprecation_msg = (
+            "class_prior/fit_prior are deprecated in 1.4 and will be removed in 1.6."
+        )
+        if self.priors is not None and (
+            self.class_prior != "deprecated" or self.fit_prior != "deprecated"
+        ):
+            raise ValueError(
+                "Cannot specify both `priors` and `class_prior`/`fit_prior`. Use only "
+                "`priors`. "
+                + deprecation_msg
+            )
+
+        if self.priors is not None:
+            self._priors = self.priors
+        else:
+            if self.class_prior != "deprecated" or self.fit_prior != "deprecated":
+                warnings.warn(deprecation_msg, FutureWarning)
+
+            if not (isinstance(self.class_prior, str) or self.class_prior is None):
+                # certainly an array-like then
+                self._priors = self.class_prior
+            elif self.fit_prior:
+                self._priors = "empirical"
+            else:
+                self._priors = "uniform"
+
+    def _update_class_log_prior(self):
```

Comment:
Comes from the `_BaseDiscreteNB` and slightly changed to use the `_priors` attribute.